crates/auth/src/lib.rs

```diff
@@ -1,2 +1,28 @@
 // SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
 // SPDX-License-Identifier: Apache-2.0
+pub mod auth;
+pub mod error;
+pub mod types;
+
+use auth::{github_login, github_login_authorized, AUTH_URL, TOKEN_URL};
+use axum::{extract::Extension, routing::get, Router};
+use oauth2::{basic::BasicClient, AuthUrl, ClientId, ClientSecret, RedirectUrl, TokenUrl};
+
+pub const COOKIE_NAME: &str = "SESSION";
+pub const GITHUB: &str = "/auth/github";
+pub const GITHUB_AUTHORIZED: &str = "/github/authorized";
+
+pub fn routes(host: &str, client_id: String, client_secret: String) -> Router {
```

Comment:
```suggestion
pub fn app(host: &str, client_id: String, client_secret: String) -> Router {
```

for consistency with other components